### ISSUE TITLE
requireAsync .then is not a function

### ISSUE THREAD

**Issue by Aid19801** [gregberge/loadable-components#601]
## 🐛 Bug Report

Implemented loadable via npm package, followed docs to try, throwing async errors in chrome console.

## To Reproduce
setup a Create React App
Create a page that is nested in App.js
Within that page's js file, make it a loadable component
Error throws.

## Expected behavior

Lazily loads in the js for that page

## Link to repl or repo (highly encouraged)

Just use CRA. 

Issues without a reproduction link are likely to stall.

## Run `npx envinfo --system --binaries --npmPackages @loadable/component,@loadable/server,@loadable/webpack-plugin,@loadable/babel-plugin --markdown --clipboard`

Paste the results here:

```
## System:
 - OS: macOS Mojave 10.14.6
 - CPU: (4) x64 Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz
 - Memory: 1.26 GB / 16.00 GB
 - Shell: 5.3 - /bin/zsh
## Binaries:
 - Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
 - Yarn: 1.21.1 - /usr/local/bin/yarn
 - npm: 6.11.3 - ~/.nvm/versions/node/v10.17.0/bin/npm
## npmPackages:
 - @loadable/component: ^5.13.1 => 5.13.1 
```

![image](https://user-images.githubusercontent.com/15033409/87932576-a40c0880-ca83-11ea-823a-01f82887d182.png)
![image](https://user-images.githubusercontent.com/15033409/87932583-aa01e980-ca83-11ea-95e7-6b331a81c2ae.png)
![image](https://user-images.githubusercontent.com/15033409/87932604-b1c18e00-ca83-11ea-8cd8-1086fa662686.png)

**Comment by theKashey** [comment]
> see https://github.com/gregberge/loadable-components/issues/428

Are you sure you've followed the docs properly?
Loadable requires a `babel plugin` for living, and CRA do not let you add one, accepting only babel macros ones, which loadable do not expose

That said - loadable is not compatible with CRA without injecting additional babel plugin.
In order to use please:
- use craco, eject or other way to add a custom babel plugin to CRA
- use [imported-component](https://github.com/theKashey/react-imported-component), which has macros
- help us implement those macros here - https://github.com/gregberge/loadable-components/issues/602

**Comment by Aid19801** [comment]
Hi @theKashey thanks for replying. I spun-up a CRA just to try a bare bones example, that would be easy to replicate.

In reality, I am trying to use `loadable` on a SSR production app. It is indeed using the babel and webpack setup from docs.

code examples - perhaps you can tell me what i'm doing wrong?

webpack.client.js

```
module.exports = (env) => {
  const IS_DEV = env.NODE_ENV !== 'production';

  const plugins = [
   ...
    new LoadablePlugin(),
   ...
  ];

 return {
    resolve: {
      alias: {
        services: path.resolve(__dirname, 'src/client/services'),
        ...(IS_DEV ? { 'react-dom': '@hot-loader/react-dom' } : {})
      }
    },
    entry: IS_DEV ? ['./src/client/index.js', 'webpack-hot-middleware/client'] : './src/client/index.js',
    output: {
    },
    optimization: {
    },
    module: {
        rules: {...
    }
    plugins: removeEmpty(plugins), // webpack-config-utils
    node: {
      net: 'empty',
      tls: 'empty',
      dns: 'empty'
    }
  };
};
}
```

my babelrc file:

```
module.exports = {
  'ignore': [
    'node_modules/**/*'
  ],
  'presets': [
    '@babel/preset-env',
    '@babel/preset-react'
  ],
  'plugins': [
    "@loadable/babel-plugin", // as per docs
    [
```

server/react.js

```
import { ChunkExtractor } from '@loadable/server';
import routesRequest from 'client/routes'; // matches routes to MySignupPage container as-an-object so static methods can be called on each for ssr.

export const renderReactRoute = async (res, url) => {
    ...
    const routes = await (IS_DEV ? require('client/routes').default : routesRequest)(); // see routes.js below
    const activeRoute = fetchComponentData(url, store, routes);
    ...
    const statsFile = path.resolve(
    process.cwd(), 'public/assets/loadable-stats.json'
  );

   ...
   
  const MyApp = (
    <Provider store={store}>
      <StaticRouter location={url} context={context}>
        {renderRoutes(routes)}
      </StaticRouter>
    </Provider>
  );

  const webExtractor = new ChunkExtractor({ statsFile }) // i ran a build and this appears now in my public file as above. Not sure if i need node one too? Why that would be missing or where it would be.

  const jsx = webExtractor.collectChunks(<MyApp />);

  const html = renderToString(jsx);
    ...
   const htmlString = template(store.getState(), config, html, Helmet.renderStatic(), res, analytics);
    ...
  return htmlString;
}

router.all('*', (req, res, next) => {
  renderReactRoute(res, req.url).then(htmlString => {
    if (!res.headersSent) {
      res.send(htmlString);
    }
  }).catch(err => {
    // console.log('Exception thrown during react server side rendering for', req.url);
    return next(err);
  });
});

```

client/index.js

```
...
const render = Component => {

  loadableReady(() => {

    // this is the only instance they have of hydrate()

    Component().then(ContainerComponent => {
      hydrate(
        <AppContainer>
          <ContainerComponent
            history={ history }
            store={ store }
          />
        </AppContainer>,
        document.getElementById('mount')
      );
    });

  })
};
...
```

then in my page container Signup.js

```
import React, { Component } from "react";
import loadable from '@loadable/component';

const AsyncPageSignup = loadable((props) => <MySignupPage {...props} />);

class MySignupPage extends Component {
...
```

Basically i just want to split this bloated Signup page out so it loads in separately. Obviously i'm doing something wrong?
